crates/components/authdecode/authdecode-core/src/prover/commitment.rs

```diff
@@ -0,0 +1,252 @@
+        assert!(encodings.len() == bit_ids.len());
+
+        let encodings = plaintext
+            .to_msb0_vec()
+            .into_iter()
+            .zip(encodings)
+            .map(|(bit, enc)| Encoding::new(*enc, bit))
+            .collect::<Vec<_>>();
+
+        CommitmentData {
+            encodings: ActiveEncodings::new(encodings, bit_ids),
+        }
+    }
+
+    /// Convert `self` into an iterator over chunks of the commitment data. If `chunk_size` does not
+    /// divide the length of the commitment data, then the last chunk will not have length `chunk_size`.
+    ///
+    /// # Arguments
+    ///
+    /// * `chunk_size` - The size of a chunk.
+    pub fn into_chunks(self, chunk_size: usize) -> CommitmentDataChunks<I> {
+        CommitmentDataChunks {
+            encodings: self.encodings.clone().into_chunks(chunk_size * 8),
+        }
+    }
+}
+
+pub struct CommitmentDataChunks<I> {
+    encodings: ActiveEncodingsChunks<I>,
+}
+
+impl<I> Iterator for CommitmentDataChunks<I>
+where
+    I: IdCollection,
+{
+    type Item = CommitmentDataChunk<I>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.encodings
+            .next()
+            .map(|encodings| Some(CommitmentDataChunk { encodings }))?
+    }
+}
+
+/// A chunk of data that needs to be committed to.
+pub struct CommitmentDataChunk<I>
+where
+    I: IdCollection,
+{
+    /// The active encoding of each bit of the plaintext. The number of encodings is always a
+    /// multiple of 8.
+    encodings: ActiveEncodings<I>,
+}
+
+impl<I> CommitmentDataChunk<I>
+where
+    I: IdCollection,
+{
+    /// Creates a commitment to this chunk.
+    #[allow(clippy::borrowed_box)]
+    fn commit<F>(&self, backend: &Box<dyn Backend<F>>) -> ChunkCommitmentDetails<I, F>
+    where
+        F: Field + Clone + std::ops::Add<Output = F>,
+    {
+        let sum = self.encodings.compute_sum::<F>();
+
+        let (plaintext_hash, plaintext_salt) = backend.commit_plaintext(self.encodings.plaintext());
+
+        let (encoding_sum_hash, encoding_sum_salt) = backend.commit_encoding_sum(sum.clone());
+
+        ChunkCommitmentDetails {
+            plaintext_hash,
+            plaintext_salt,
+            encodings: self.encodings.clone(),
+            encoding_sum: sum,
+            encoding_sum_hash,
+            encoding_sum_salt,
+        }
+    }
+
+    /// Creates a commitment to this chunk with the provided salt.
+    #[allow(clippy::borrowed_box)]
+    fn commit_with_salt<F>(
+        &self,
+        backend: &Box<dyn Backend<F>>,
+        salt: F,
+    ) -> ChunkCommitmentDetails<I, F>
+    where
+        F: Field + Clone + std::ops::Add<Output = F>,
+    {
+        let sum = self.encodings.compute_sum::<F>();
+
+        let plaintext_hash =
+            backend.commit_plaintext_with_salt(self.encodings.plaintext(), salt.clone());
+
+        let (encoding_sum_hash, encoding_sum_salt) = backend.commit_encoding_sum(sum.clone());
```

Comment:
yes, there is no way since the salt for the encoding sum is not gonna be used later, it is only used in the Authdecode protocol, whereas the plaintext salt **will** be used later when creating zkps.